src/basic/rm-rf.h

```diff
@@ -12,9 +12,17 @@ typedef enum RemoveFlags {
         REMOVE_PHYSICAL         = 1 << 2, /* If not set, only removes files on tmpfs, never physical file systems */
         REMOVE_SUBVOLUME        = 1 << 3, /* Drop btrfs subvolumes in the tree too */
         REMOVE_MISSING_OK       = 1 << 4, /* If the top-level directory is missing, ignore the ENOENT for it */
-        REMOVE_CHMOD            = 1 << 5, /* chmod() for write access if we cannot delete something */
+        REMOVE_CHMOD            = 1 << 5, /* chmod() for write access if we cannot delete or access something */
+        REMOVE_RESTORE_MODE     = 1 << 6, /* Restore the old mode before returning */
```

Comment:
mhmm, maybe rename this to REMOVE_CHMOD_RESTORE to make clear it belongs to REMOVE_CHMOD